game/src/core/utils/IVector2.java

```diff
@@ -0,0 +1,254 @@
+   *
+   * @param other The other vector.
+   * @return A new vector that is the difference of this and the other vector.
+   */
+  default IVector2 subtract(IVector2 other) {
+    return IVector2.of(x() - other.x(), y() - other.y());
+  }
+
+  /**
+   * Multiplies this vector by a scalar.
+   *
+   * @param scalar The scalar.
+   * @return A new vector that is this vector multiplied by the scalar.
+   */
+  default IVector2 scale(float scalar) {
+    return IVector2.of(x() * scalar, y() * scalar);
+  }
+
+  /**
+   * Multiplies this vector by two scalars.
+   *
+   * @param scalarX The scalar for the x component.
+   * @param scalarY The scalar for the y component.
+   * @return A new vector that is this vector multiplied by the scalars.
+   */
+  default IVector2 scale(float scalarX, float scalarY) {
+    return IVector2.of(x() * scalarX, y() * scalarY);
+  }
+
+  /**
+   * @return The length of the vector.
+   */
+  default float length() {
+    return (float) Math.sqrt(x() * x() + y() * y());
+  }
+
+  /**
+   * Normalizes the vector to a unit vector (length of 1).
+   *
+   * <p>Normalization is useful for ensuring that the vector maintains its direction but has a
+   * length of 1.
+   *
+   * <p>If the vector is zero (length 0), it returns the zero vector.
+   *
+   * @return A new vector that is the normalized version of this vector or the zero vector if the
+   *     length is 0.
+   */
+  default IVector2 normalize() {
+    float len = length();
+    if (len == 0) {
+      return IVector2.ZERO;
+    }
+    return IVector2.of(x() / len, y() / len);
+  }
+
+  /**
+   * Computes the dot product of this vector and another vector. The dot product is useful for
+   * determining the angle between vectors and projecting one vector onto another.
+   *
+   * @param other The other vector
+   * @return The dot product as a scalar value
+   */
+  default float dot(IVector2 other) {
+    return x() * other.x() + y() * other.y();
+  }
+
+  /**
+   * Calculates the squared length of the vector. This is more efficient than length() when you only
+   * need to compare distances or when the actual length value isn't needed.
+   *
+   * @return The squared length of the vector
+   */
+  default float lengthSquared() {
+    return x() * x() + y() * y();
+  }
+
+  /**
+   * Calculates the distance between this vector and another vector.
+   *
+   * @param other The other vector
+   * @return The distance between the two vectors
+   */
+  default float distance(IVector2 other) {
+    return subtract(other).length();
+  }
+
+  /**
+   * Rotates this vector by a specified angle in degrees.
+   *
+   * @param degrees The angle in degrees to rotate the vector.
+   * @return A new vector that is the result of rotating this vector by the specified angle.
+   */
+  default IVector2 rotateDeg(float degrees) {
+    float radians = (float) Math.toRadians(degrees);
+    float cos = (float) Math.cos(radians);
+    float sin = (float) Math.sin(radians);
```

Comment:
warum machst du hier explizite konvertierungen? würde es sinn machen, hier mit den werten weiter zu rechnen und einen weiteren "konstruktor" `of(double x, double y)` anzubieten?